dataframe-excel/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/xlsx.kt

```diff
@@ -470,6 +471,26 @@ private fun Cell?.cellValue(sheetName: String): Any? {
     return getValueFromType(cellType)
 }
 
+public enum class WorkBookType {
+    XLS,
+    XLSX,
+}
+
+/**
+ * Writes this DataFrame to an Excel file as a single sheet.
+ *
+ * Implemented with [Apache POI](https://poi.apache.org) using `HSSFWorkbook` for XLS files,
+ * `XSSFWorkbook` for standard XLSX files, and `SXSSFWorkbook` for memory-efficient streaming when creating new XLSX files.
+ *
+ * @param path The path to the file where the data will be written.
+ * @param columnsSelector A [selector][ColumnsSelector] to determine which columns to include in the file. The default is all columns.
+ * @param sheetName The name of the sheet in the Excel file. If null, the default name will be used.
+ * @param writeHeader A flag indicating whether to write the header row in the Excel file. Defaults to true.
+ * @param workBookType The [type of workbook][WorkBookType] to create (e.g., XLS or XLSX). Defaults to XLSX.
+ * @param keepFile If `true` and the file already exists, a new sheet will be appended instead of overwriting the file.
```

Comment:
if sheetName already exists in the workbook, will it be overwritten? maybe comment needs to be adjusted

Reply:
Throws:
IllegalArgumentException – if the name is null or invalid or workbook already contains a sheet with this name

Reply:
What's the best thing to write in our doc?

Reply:
hmm.. maybe `@throws`